src/mlpack/core/data/scaler_methods/maxabsscaler.hpp

```diff
@@ -0,0 +1,97 @@
+/**
+ * @file maxabsscaler.hpp
+ * @author Jeffin Sam
+ *
+ * MaxAbsScaler class to scale features.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_DATA_MAX_ABS_SCALE_HPP
+#define MLPACK_CORE_DATA_MAX_ABS_SCALE_HPP
+
+#include <mlpack/prereqs.hpp>
+
+namespace mlpack {
+namespace data {
+
+/**
+ * A simple MaxAbs Scaler class.
+ *
+ * Given an input dataset this class helps you to scale each
+ * feature by its maximum absolute value.
+ *
+ * @code
+ * arma::Mat input = loadData();
+ * arma::Mat output;
+ *
+ * // Scale the features.
+ * MaxAbsScaler scale;
+ * scale.Tranform(input, output);
+ *
+ * // Retransform the input.
+ * scale.InverseTransform(output, input);
+ * @endcode
+ */
+class MaxAbsScaler
+{
+ public:
+  /**
+  * Function to scale Features.
+  *
+  * @param input Dataset to scale features.
+  * @param output Output matrix with scaled features.
+  */
+  template<typename MatType>
+  void Transform(const MatType& input, MatType& output)
+  {
+    output.copy_size(input);
+    itemMin = arma::min(input, 1);
+    itemMax = arma::max(input, 1);
+    scale = arma::max(arma::abs(itemMin), arma::abs(itemMax));
+    // Handling zeros in scale vector.
+    for (size_t i = 0; i < scale.n_elem; i++)
+    {
+      if (scale(i) == 0)
+      {
+        scale(i) = 1;
+      }
+    }
+    output = input.each_col() % (1.0 / scale);
```

Comment:
It might make the code a bit longer but you might consider iterating over columns and then multiplying by `1.0 / scale` only if `scale(i) != 0`.  (Also, you can just do `/ scale` instead of `% (1.0 / scale)`.)

Another comment, should we center the data here too?

Reply:
Hehe, I see this comment actually contradicts Marcus's comment.  Up to you what you'd like to do.

Reply:
> Hehe, I see this comment actually contradicts Marcus's comment. Up to you what you'd like to do.

I have made minor changes, but sticking to it

Reply:
> It might make the code a bit longer but you might consider iterating over columns and then multiplying by `1.0 / scale` only if `scale(i) != 0`. (Also, you can just do `/ scale` instead of `% (1.0 / scale)`.)

Adapted changes , `/ scale`
 
> Another comment, should we center the data here too?

Not sure, whether  it would be a good choice , I have implemented the standard version which didn't involve any centering of data

The standard[ implementation](https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.MaxAbsScaler.html#sklearn.preprocessing.MaxAbsScaler)

Reply:
Sounds good to me, we can leave scaling out here. :+1: Thanks for the fixes.